examples/sbol2/CreLoxRecombination.ipynb

```diff
@@ -0,0 +1,184 @@
+    "\n",
+    "# Constraint to ensure that CDS and loxP2 are adjacent\n",
+    "constraint2 = recombination_system.sequenceConstraints.create('CDS_to_loxP2')\n",
+    "constraint2.subject = cds_comp.identity\n",
+    "constraint2.object = loxP2_comp.identity\n",
+    "constraint2.restriction = sbol2.SBOL_RESTRICTION_PRECEDES"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "'Valid.'"
+      ]
+     },
+     "execution_count": 7,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "# Validate the document to ensure compliance with SBOL standards\n",
+    "doc.validate()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "'Valid.'"
+      ]
+     },
+     "execution_count": 8,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "# Save the document to an SBOL file\n",
+    "doc.write('cre_lox_recombination.xml')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Explanation\n",
+    "\n",
+    "In this example, we modeled a Cre-Lox recombination system:\n",
+    "\n",
+    "1. **Cre-Lox Recombination System**: We defined a `ComponentDefinition` to represent the entire recombination system. This component holds the loxP sites and the unknown CDS as its sub-components.\n",
+    "   \n",
+    "2. **loxP Sites**: We created two `Component` objects representing the loxP recombination sites. Each loxP site is linked to a known DNA sequence and added as a component to the system.\n",
+    "\n",
+    "3. **Unknown CDS**: We defined an unknown or not-yet-selected coding sequence (CDS) using `GenericLocation`, which allows us to keep the boundaries flexible and represents the CDS in reverse complement orientation. The CDS is also added as a component to the system.\n",
+    "\n",
+    "4. **SequenceAnnotations**: We created `SequenceAnnotations` for each element (loxP sites and CDS), using a `Range` for the first loxP site and `GenericLocation` for the CDS and the second loxP site, enabling flexible positioning and orientation.\n",
+    "\n",
+    "5. **SequenceConstraints**: Two `SequenceConstraint` objects were added to ensure that the CDS is flanked by the loxP sites in the correct order. These constraints enforce that:\n",
+    "   - The first loxP site precedes the CDS.\n",
+    "   - The CDS precedes the second loxP site.\n",
+    "\n",
+    "This setup simulates a Cre-Lox recombination event, where the recombination between the two loxP sites would flip the unknown CDS into the correct orientation, potentially activating it. By modeling the system this way, we maintain flexibility in defining the sequence, structure, and relationships between components."
```

Comment:
```suggestion
    "This setup simulates a Cre-Lox recombination device, where the recombination between the two loxP sites would flip the unknown CDS into the correct orientation, potentially activating it. By modeling the system this way, we maintain flexibility in defining the sequence, structure, and relationships between components."
```

We aren't actually modeling the event here - for that, we'd need a `ModuleDefinition`, which is a pain to work with in SBOL2.